ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -369,6 +374,15 @@ class SimpleOperationTracker implements OperationTracker {
         originatingDcName);
   }
 
+  /**
+   * @param eligibleReplicaCount eligible replicas to send Delete Requests.
+   * @return true if it's possible to run offline repair for the Delete
+   */
+  public boolean possibleRunOfflineRepair(int eligibleReplicaCount) {
+    return (eligibleReplicaCount >= 1 && routerOperation == RouterOperation.DeleteOperation
+        && routerConfig.routerDeleteOfflineRepairEnabled && routerConfig.routerRepairRequestsDbFactory != null);
```

Comment:
Would it be safer if we check getRepairRequestsDb is not null? Since it might failed to create the db.